pwiz_tools/Skyline/FileUI/PeptideSearch/SearchControl.cs

```diff
@@ -142,8 +143,11 @@ protected void UpdateSearchEngineProgress(IProgressStatus status)
             }
 
             // look at the last 10 lines for the same message and if found do not relog the same message
-            if (_progressTextItems.Skip(Math.Max(0, _progressTextItems.Count - 10)).Any(entry => entry.Message == message))
```

Comment:
It seems this blatant missing optimization has been in .NET since 2016 or so, but alas doesn't seem to have made it into Framework.